src/dmd/root/array.d

```diff
@@ -253,3 +253,63 @@ private:
     size_t len;
     size_t *ptr;
 }
+
+/**
+ * Converts the given root Array to the standard D array.
+ * Params:
+ *  array = the array to convert.
+ * Returns:
+ *  The given array converted to a standard D array.
+ */
+@property T[] toDArray(T)(ref Array!T array)
```

Comment:
This seems to only be here to support the unittest, but it is a public symbol.  Please embed it in the unittest or encase it in `version(unittest)`.  Also, I don't think it should be a property; I recommend making it a function (functions do have optional parens in D).

Reply:
The only usage currently is in the unittest, however, it seemed like a reasonable function to have available.

Reply:
What criteria do you use to determine when a function should be a property?

Reply:
When you want to create an abstraction of stored state.  This looks more like an operation to me.

Reply:
I wouldn't consider this an operation at all.  Just an abstraction of the underlying Array structure.  A "different way" of looking at the Array.

Reply:
Maybe you would prefer `asDArray` so it sounds more like an abstraction and less like an operation?

Reply:
I see you are using it more as an extension property (C# terminology), so I guess that make sense, but it's presence in the module implies that it's exposing some state in the `array` module, so it's a little weird.  If you insist on keeping it as a property, then yes a name that doesn't start with a verb or preposition implying an action would be appropriate.  I suggest `DArray`, but I'll go along with `asDArray` as at least it's better than `toDArray`.  But I'm not going to be too picky about it.  I've had my say.  You're the author doing the work, so you make the decision.

Reply:
The naming convention in Phobos is to use `to` if allocation happens and `as` for new views or lazy ranges. Std.uni makes a good precedent in this regard.